modules/regional-go-service/main.tf

```diff
@@ -38,13 +38,25 @@ resource "ko_build" "this" {
   importpath  = each.value.source.importpath
 }
 
-// Sign each of the application images.
 resource "cosign_sign" "this" {
   for_each = var.containers
   image    = ko_build.this[each.key].image_ref
   conflict = "REPLACE"
 }
 
+locals {
+  // Pull out the containers from var.containers that has no port
+  sidecars = {
+    for key, value in var.containers : key => value if length(value.ports) == 0
+  }
+  // Pull out the main container from var.containers that has a port.
+  // There should be only one of them, but using a map to make it easier to
+  // iterate over and look up the ko_builds.
+  main_container = {
```

Comment:
do we want at check to see length(main_container) == 1?

we dont want this greater than one.

Reply:
Done